lib/ansible/modules/windows/win_get_url.ps1

```diff
@@ -155,28 +332,82 @@ Function Download-File($module, $url, $dest, $headers, $credentials, $timeout, $
         }
     }
 
-    if (-not $module.CheckMode) {
-        # FIXME: Single-out catched exceptions with more specific error messages
-        Try {
-            $extWebClient.DownloadFile($url, $dest)
-        } Catch [System.Net.WebException] {
-            $module.Result.status_code = [int] $_.Exception.Response.StatusCode
-            $module.Result.elapsed = ((Get-Date) - $module_start).TotalSeconds
-            $module.FailJson("Error downloading '$url' to '$dest': $($_.Exception.Message)", $_)
-        } Catch {
-            $module.Result.elapsed = ((Get-Date) - $module_start).TotalSeconds
-            $module.FailJson("Unknown error downloading '$url' to '$dest': $($_.Exception.Message)", $_)
+    Try {
+
+        $tmpDest = [System.IO.Path]::GetTempFileName()
+        $diff += "+$tmpDest`n"
+        $extWebClient.DownloadFile($url, $tmpDest)
+
+        $tmpDestHash = Get-FileChecksum -path $tmpDest -algorithm $checksum_algorithm
+
+        # Checksum verification for downloaded file
+        if ($checksum) {
+            # Check both hashes are the same
+            if ($tmpDestHash -ne $checksum) {
+                throw [string]("The checksum for {0} did not match '{1}', it was '{2}'" -f $dest, $checksum, $tmpDestHash)
+            } else {
+                if(Test-Path -LiteralPath $dest) {
+                    $destHash = Get-FileChecksum -path $dest -algorithm $checksum_algorithm
+                    if ($destHash -eq $checksum) {
+                        $module.Result.status_code = 200
+                        $module.Result.msg = 'file already exists'
```

Comment:
We seem to be setting `status_code`, `msg` in a few different places and it relates to the web request output. This seems to be a common theme that should be handled in the 1 location.

Reply:
Could kindly suggest some changes?
I have no idea how to implement you advice.
Now it works and passes all tests.

Reply:
@jborean93 Could you please take a look?

Reply:
This goes against our documentation and we don't want to merge changes that break this. We also don't want to make changes that will make this module harder to maintain in the future.

I don't see how/why we should be using the `msg` return value for our checks of idempotency.

Reply:
Fixed